### PR TITLE
Describe the project in the readme

### PR DESCRIPTION
**Pull request by thayeylolu**
- [X] Project's About Header
- [X] Column description
- [X] Added Tiff's template

### CODE REVIEW

**Comment by liannah** [comment]
I am not very sure about this sentence:  Although its original creators are unknown, previous studies strongly suggest it is a private data. Can you elaborate a little more on what do you mean by private, also I think the authors are from UCI thus no need to mention them by names for them to be known. 

Also I will change the X- format naming, as in our csv file the columns have different names, please take a look and instead of saying Y -> say the name of the column and in the description specify that it is our response variable. As for the explanation I will add for "Y": yes if the person will do default payment next month, no otherwise. Let me know if you think this interpretation is not correct.  @thayeylolu

**Comment by thayeylolu** [comment]
Thanks @liannah

**Comment by liannah** [comment]
Oh i see. In the csv file the first row of columns were dropped. and only the second row of columns were used. I am not sure if we should use xls file instead and drop the first column or just use csv version of xls ? May be we can discuss this part during our meeting or @karanpreetkaur can weigh in as she is the one taking care of py file. 

Also, is this a final version ? I think we can write a little more in the `about` section. But I think you can merge it like this as a draft and others can weight in as well. @thayeylolu

**Comment by Arushi282** [comment]
Hi @thayeylolu, I think your approach to the readme file for this milestone is based on the wrong template. We should take inspiration from the template in milestone 1 release of Tiffany's project and also on reading the milestone in depth again, most of the project proposal will reside in the README file, so most of the points covered in the Part 3 of the milestone 1 assignment should be addressed here. Below I have provided some things that definitely need to be covered. 

(1) However, I do really like the detail provided on columns of the dataset, I think we can remove the X1, X2 stuff and just describe the columns we are working with for this prediction. 

(2) Firstly, could please add everyones name in on the top (before about), you can do something like 
authors: <add name>,
                <add name> 

(3) What is our predictive question: I  really like how you phrased in the eda summary: 
"This project aims to build a classification model using a machine learning algorithm to predict potential credit default 
 accounts of Taiwan's credit card clients'. " From tiffany's readme in the first milestone, our question should be 
  something like: "given past payment history of clients, can we predict credit default accounts?" . I think we should also mention potential models we should use for prediction, of the top of my head the prediction models would be Decision Tree classifier, Knn, RBM SVF. We don't need to decide now, but we it would be good to mention the ones we are considering. 

(4) Good to explain why out classification prediction and analysis is important. 

(5) Need to describe the steps in detail of how we are going about our analysis. How will we present our data finally. 

(6) Attach the report for the eda in the readme file. 

Would request you to make these changes as soon as possible.

**Comment by Arushi282** [comment]
Also I agree with Lianna, if the original creditors are not mentioned then we can just leave it at source UCI, so this sentence should be good on its own: "The data set is a Taiwan credit card data from April to September, 2005 sourced from the UCI machine learning repository and can be found here."

**Comment by karanpreetkaur** [comment]
Hi @thayeylolu 

I agree with the changes mentioned by @liannah and @Arushi282. Just few things to add from my side:

1.  If you can change default payment  with default payment next month in **- Y: The default payment . 0 represents yes and 1 represents no.** as we are predicting defaults for next month. 

2. The License section is provided on https://archive-beta.ics.uci.edu/ml/datasets/default+of+credit+card+clients for the dataset (right section) . You can directly copy it and add here.

3. For references section, I have found it from acknowledgment section of same dataset uploaded on kaggle https://www.kaggle.com/uciml/default-of-credit-card-clients-dataset

          Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml]. Irvine, CA: University of California, School of Information and Computer Science.
          
          The original dataset can be found here {--add dataset link here} at the UCI Machine Learning Repository.

   You can add above text in reference section.

**Comment by thayeylolu** [comment]
> Hi @thayeylolu, I think your approach to the readme file for this milestone is based on the wrong template. We should take inspiration from the template in milestone 1 release of Tiffany's project and also on reading the milestone in depth again, most of the project proposal will reside in the README file, so most of the points covered in the Part 3 of the milestone 1 assignment should be addressed here. Below I have provided some things that definitely need to be covered.
> 
> (1) However, I do really like the detail provided on columns of the dataset, I think we can remove the X1, X2 stuff and just describe the columns we are working with for this prediction.
> 
> (2) Firstly, could please add everyones name in on the top (before about), you can do something like authors: ,
> 
> (3) What is our predictive question: I really like how you phrased in the eda summary: "This project aims to build a classification model using a machine learning algorithm to predict potential credit default accounts of Taiwan's credit card clients'. " From tiffany's readme in the first milestone, our question should be something like: "given past payment history of clients, can we predict credit default accounts?" . I think we should also mention potential models we should use for prediction, of the top of my head the prediction models would be Decision Tree classifier, Knn, RBM SVF. We don't need to decide now, but we it would be good to mention the ones we are considering.
> 
> (4) Good to explain why out classification prediction and analysis is important.
> 
> (5) Need to describe the steps in detail of how we are going about our analysis. How will we present our data finally.
> 
> (6) Attach the report for the eda in the readme file.
> 
> Would request you to make these changes as soon as possible.

Thanks for noticing!

**Comment by thayeylolu** [comment]
Should I remove the table listing the columns @liannah @Arushi282  @karanpreetkaur ?

**Comment by karanpreetkaur** [comment]
According to me, we can remove the table as it is redundant to explain columns both in table and then text below.

**Comment by thayeylolu** [comment]
Pushed update.

**Review by karanpreetkaur — APPROVED** [review]
looks good to me. Thanks @thayeylolu  

Approved

**Review by Arushi282 — CHANGES_REQUESTED** [review]
(1) When you are defining the target class, you are saying that 0 represent yes and 1 represents no, but its actually the other way around. 

(2) I think it should be "Will a customer default a credit payment?"  not "Will a customer default  credit payment?". 

(3) Instead of saying "final" report, say for now the EDA report can be found here. 

Other than that, looks good to me, thanks in advance!

File: README.md [hunk 1]
```diff
@@ -64,28 +64,36 @@ The columns are further defined below:
...
 The final report can be found [here](https://github.com/UBC-MDS/credit_default_prediction/blob/main/src/credit_default_eda.ipynb)
 
 ## Usage
+To replicate the analysis, clone this GitHub repository, install the dependencies listed below, and run the following commands at the command line/terminal from the root directory of this project:
 
+`python src/download_data.py --out_type=xls --url=https://archive.ics.uci.edu/ml/machine-learning-databases/00350/default%20of%20credit%20card%20clients.xls --out_file=data/raw/default_credit_card_clients.csv`
 
 ## Dependencies
-
+- Python 3.9 and Python packages:
```
[the remark this answers is not in the document]
Comment (liannah) [reply README.md:81]: add altair-data-server=0.4.1 
       portpicker=1.5.0

**Review by liannah — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -64,28 +64,36 @@ The columns are further defined below:
...
+Finally, we will re-fit the training data set on our final model then evaluate the performance on the test data set on accuracy and alternative scoring/evaluation metrics such as f1-score or recall. These values will be reported as a table in the final report.
 
 ## Report
 
 The final report can be found [here](https://github.com/UBC-MDS/credit_default_prediction/blob/main/src/credit_default_eda.ipynb)
 
 ## Usage
+To replicate the analysis, clone this GitHub repository, install the dependencies listed below, and run the following commands at the command line/terminal from the root directory of this project:
 
+`python src/download_data.py --out_type=xls --url=https://archive.ics.uci.edu/ml/machine-learning-databases/00350/default%20of%20credit%20card%20clients.xls --out_file=data/raw/default_credit_card_clients.csv`
```
[the remark this answers is not in the document]
Comment (thayeylolu) [reply README.md:78]: I do not quite understand why we are adding `jupyter lab src/credit_default_eda.ipynb`.

**Review by liannah — APPROVED** [review]

**Review by Arushi282 — APPROVED** [review]

**Comment by thayeylolu** [comment]
I have detected a syntax error. Made necessary adjustment

**Pull request merged**